include/board.h

```diff
@@ -37,8 +37,31 @@
 #include <stdbool.h>
 #include <stdint.h>
 
+#include "state_manager.h"
+
 namespace rosflight_firmware
 {
+struct debug_info_t{
```

Comment:
Also, thinking into the future, this debug info struct is very specific to F4 boards; it kind of breaks the hardware abstraction and wouldn't apply if someone creates a new board support layer. Is there a way we can make it more generic? Maybe it's just a fixed-size array of `uint32_t` at the board layer, and then it needs to be reinterpreted based on what board it's actually running on? That keeps the abstraction cleaner but at the expense of it being a little more annoying for developers. This really shouldn't be used much though, and will only be used by people who have the background to understand how to interpret it. Thoughts?

Reply:
Most of this information is currently unused, as it is not very useful for debugging.  I did try to write it so that the backup_data_t was easy to modify, so any future additions/subtractions should be pretty seamless.

Reply:
Maybe I'm missing something, but I don't see any F4-specific code here - (Every processor has a program counter, link register, etc...)

Reply:
Okay cool, I'm good with that. I'll probably expose the clearing of the backup data in another merge request, but otherwise this is pretty great!